phoenix-core/src/main/java/org/apache/phoenix/compile/ExpressionCompiler.java

```diff
@@ -361,7 +362,8 @@ protected ColumnRef resolveColumn(ColumnParseNode node) throws SQLException {
         // Disallow explicit reference to salting column, tenant ID column, and index ID column
         if (pkPosition >= 0) {
             boolean isSalted = table.getBucketNum() != null;
-            boolean isMultiTenant = context.getConnection().getTenantId() != null && table.isMultiTenant();
+            boolean isMultiTenant = context.getConnection()==null ? table.isMultiTenant() :
```

Comment:
When I parse expressions in IndexMaintainer (line 331), I first use SQLParser.parseCondition to create a parse node and then create an ExpressionCompiler and create the expression.
 At this point, I don't have access to a connection so I set it to null. The connection was only being used to get the tenant id.